src/gui/progressbarpainter.cpp

```diff
@@ -26,49 +26,43 @@
  * exception statement from your version.
  */
 
-#include "progressbardelegate.h"
+#include "progressbarpainter.h"
 
-#include <QApplication>
-#include <QModelIndex>
 #include <QPainter>
+#include <QPalette>
 #include <QStyleOptionViewItem>
 
 #if (defined(Q_OS_WIN) || defined(Q_OS_MACOS))
 #include <QProxyStyle>
 #endif
 
-ProgressBarDelegate::ProgressBarDelegate(const int progressColumn, const int dataRole, QObject *parent)
-    : QStyledItemDelegate {parent}
-    , m_progressColumn {progressColumn}
-    , m_dataRole {dataRole}
+ProgressBarPainter::ProgressBarPainter()
 {
 #if (defined(Q_OS_WIN) || defined(Q_OS_MACOS))
-    m_dummyProgressBar.setStyle(new QProxyStyle {"fusion"});
+    static QProxyStyle style {"fusion"};
+    m_dummyProgressBar.setStyle(&style);
 #endif
-}
 
-void ProgressBarDelegate::initProgressStyleOption(QStyleOptionProgressBar &option, const QModelIndex &index) const
-{
-    option.text = index.data().toString();
-    option.progress = static_cast<int>(index.data(m_dataRole).toReal());
-    option.maximum = 100;
-    option.minimum = 0;
-    option.textVisible = true;
+    m_progressBarStyle.initFrom(&m_dummyProgressBar);
```

Comment:
view item options can change dynamically.

Reply:
Which option could change? Look at the previous code, `m_dummyProgressBar` isn't changed/updated anywhere.

Reply:
it can be internally changed, forex - on macos - dark and light theme switch, and there can be other user setting changes.

Reply:
Reverted.